### PR TITLE
Add framework device token sync API

### PR DESCRIPTION
**Pull request by mikepitre**
## Summary

Adds a ClerkKit SPI for framework integrations that need to hand an existing Clerk device token into the native SDK and reconcile native client state.

This is needed by `@clerk/expo` on iOS. Expo can have two Clerk runtimes in the same app: the JS SDK and the native ClerkKit SDK used by native UI. If the user signs in through JS first, JS owns the signed-in client token while ClerkKit may still have an older cached native client, often an anonymous client. The Expo bridge needs a supported way to pass that token to ClerkKit without writing ClerkKit keychain internals directly.

Paired Expo PR: https://github.com/clerk/javascript/pull/8851

## What changed

- Adds `@_spi(FrameworkIntegration)` APIs on `Clerk`:
  - `deviceToken`
  - `updateDeviceToken(_:)`
  - `DeviceTokenError.emptyToken`
- `updateDeviceToken(_:)` stores the supplied token, clears cached client/environment state when the token changes, and refreshes the client.
- The refresh after a token replacement omits `x-clerk-client-id` for that one request so a stale cached client id cannot conflict with the newly stored device token.
- Keeps normal device-token response handling behavior: response middleware still logs keychain write failures without failing the response.
- Adds focused tests for token update, cache clearing, blank token rejection, and omitting the client id header.

## Notes

The API is intentionally SPI rather than public API. Device tokens are SDK integration credentials, not the token app developers should use for backend authentication. Keeping this behind `FrameworkIntegration` gives wrapper SDKs a supported integration point without making device-token handling part of the general ClerkKit public surface.

## Testing

```sh
swift test --filter ClientTests --filter ClerkHeaderRequestMiddlewareTests
```

Passed: 24 tests in 3 suites.

### CODE REVIEW

File: Sources/ClerkKit/Core/Clerk+DeviceToken.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  public func updateDeviceToken(_ token: String) async throws -> Client? {
+    guard !token.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
+      throw DeviceTokenError.emptyToken
+    }
+
+    let previousToken = deviceToken
+    try storeDeviceToken(token)
+
+    if previousToken != token {
+      clearCachedClientStateAfterDeviceTokenChange()
```
[the remark this answers is not in the document]
Comment (mikepitre) [reply Sources/ClerkKit/Core/Clerk+DeviceToken.swift]: Fixed in `85589ebe`.

I added a `ClientResponseGeneration` boundary for device-token swaps. Requests capture the current generation when prepared, and `updateDeviceToken` advances the generation when cached client state is cleared. Client responses from an older generation are ignored in both the response middleware path and the direct `refreshClient()` path, so an in-flight request from the old token/client cannot restore stale native state after the token swap.

Added coverage for request tagging, response middleware stale-generation handling, and direct `refreshClient()` stale-generation handling.

Verified with:

```sh
swift test --filter ClientTests --filter ClerkHeaderRequestMiddlewareTests --filter ClerkClientSyncResponseMiddlewareTests
```

`35 tests in 4 suites` passed.

[the remark this answers is not in the document]
Comment (mikepitre) [reply Sources/ClerkKit/Core/Clerk+DeviceToken.swift:54]: Fixed in `329706dd`.

`updateDeviceToken(_:)` now trims whitespace/newlines once and uses the normalized token for validation, storage, and comparison. I also updated the token update test to pass a token with surrounding whitespace and assert that the stored device token is normalized.

Verified with:

```sh
swift test --filter ClientTests --filter ClerkHeaderRequestMiddlewareTests --filter ClerkClientSyncResponseMiddlewareTests
```

`35 tests in 4 suites` passed.

[the remark this answers is not in the document]
Comment (mikepitre) [reply Sources/ClerkKit/Core/Clerk+DeviceToken.swift]: @codex does the fix commit address your concern?

**Comment by mikepitre** [comment]
@codex review again

File: Sources/ClerkKit/Networking/Middleware/ClerkDeviceTokenResponseMiddleware.swift [hunk 2]
```diff
@@ -15,7 +15,11 @@ struct ClerkDeviceTokenResponseMiddleware: ClerkResponseMiddleware {
   func validate(_ response: HTTPURLResponse, data _: Data, for _: URLRequest) async throws {
     if let deviceToken = response.value(forHTTPHeaderField: "Authorization") {
       try await runtimeScope.withCurrentClerk {
-        $0.storeReceivedDeviceToken(deviceToken)
+        do {
+          try $0.storeDeviceToken(deviceToken)
```
[the remark this answers is not in the document]
Comment (mikepitre) [reply Sources/ClerkKit/Networking/Middleware/ClerkDeviceTokenResponseMiddleware.swift:28]: Fixed in `48a21e3b`.

The device-token response middleware now checks the request client-response generation before storing an `Authorization` header. If the request was prepared before a device-token generation change, the stale header is ignored, matching the client-sync guard already added for stale response bodies.

Added `ClerkDeviceTokenResponseMiddlewareTests` coverage for both the normal header-store path and the stale-generation skip.

Verified with:

```sh
swift test --filter ClerkDeviceTokenResponseMiddlewareTests --filter ClerkClientSyncResponseMiddlewareTests --filter ClerkHeaderRequestMiddlewareTests --filter ClientTests
```

`37 tests in 5 suites` passed.

**Comment by mikepitre** [comment]
@codex review again

**Review by seanperez29 — APPROVED** [review]
